File: RodCutting/RodCutting.py
class Solution:
    def solve(self, prices, n):
        dp = [0 for i in range(n+1)]
        
        for i in range(1,n+1):
            for j in range(len(prices)):
                if j < i:
                    dp[i] = max(dp[i], dp[i-j-1] + prices[j])
        
        return dp[n]
        

File: RodCutting/RodCutting.java
import java.util.*;

class Solution {
    public int solve(int[] prices, int n) {
        int dp[] = new int[n+1];

        for (int i = 1; i <= n; i++) {
            for (int j = 0; j < prices.length; j++) {
                if (j < i) {
                    dp[i] = Math.max(dp[i], dp[i-j-1] + prices[j]);
                }
            }
        }

        return dp[n];
    }
}
